### DISCUSSION TITLE
💡 Rule Ideas

### DISCUSSION THREAD

**Discussion by timdeschryver** [timdeschryver/eslint-plugin-ngrx#134] [Ideas]
👋 Hi there,

If you have an idea for a rule, feel free to add a comment so we can discuss the idea and take the next steps.

**Comment by rafaelss95** [comment]
Not sure if there's a smart way to detect the _old_ way of action creation + `reducer` switch-based, but it'd be good to have this.

I can only see we checking the classes that implements `Action` (from `@ngrx/store`), the problem, of course, is that it only works if a class implements the `Action`. 

For the reducer, I can't see a way.

Related: https://github.com/timdeschryver/eslint-plugin-ngrx/issues/223#issuecomment-930597232

**Reply by timdeschryver** [reply]
@rafaelss95 Are you proposing a rule to report classes that implement from  `Action`?

**Reply by rafaelss95** [reply]
Yeah, but at the same time I see how fragile it could be as someone could not implement `Action` and still use "Action classes".

**Reply by timdeschryver** [reply]
I wasn't a big fan to add a rule for this to be honest.
But maybe since it has been released for a while it makes sense.
The rule could prevent someone to create an "old" action while the rest of the team is using creators.

I think that in most cases, `Action` is used though. That shouldn't be our concern.
